### PR TITLE
docs-43 - setup navigation

### PR DESCRIPTION
**Pull request by ljsinclair**
Here's the nav setup for the redesign

### CODE REVIEW

**Comment by gthrone25** [comment]
@ljsinclair Putting notes below as they mostly relate to structure at this point:
1. Great start! thanks for getting this out so quick
2. I noticed the welcome page changed. I think @kordless was pretty adamant on that being the page it is today, so I'll let you two hash that one out :) 
3. I think you can get rid of the Enterprise path and just have those same folders for Community. Looking for feedback from @GRaska , @jrbrinlee1 , and @kcrodgers24 here, but it seems like we don't need a whole section for enterprise.
4. I don't see anything under Concepts as of now, but we'll need that for pages today. Will that mirror cloud and community?
5. Cloud and Community seem to have different folder, but likely should be the same right? (i.e. Enterprise is missing data modeling, and cloud is missing API, etc.)
6.  SQL-Preview should come out of enterprise and go where SQL Guide is. We might have a Sql preview expand and a SQL expand. Most of the pages under SQL guide are actually a part of the preview (those have a note at the tope of the page)
7. There are a lot of pages missing right now, but every page should have a home before we can merge in. I know you just started this but just as a heads up. Let me know if you need help sorting
8. Once pages are there, links are.... DEAD. Drives the importance of let's get everything in the right place, then go in and address links. I'll help gather external links, but you can use the python script in the repo (dead link seeker) to find all the dead links. There are instructions in the README, but let me know if ya need some help on that.

Sorry for the long comment, but hopefully this gets any other feedback out there! We'll keep iterating!

**Comment by ljsinclair** [comment]
> @ljsinclair Putting notes below as they mostly relate to structure at this point:
> 
> 1. Great start! thanks for getting this out so quick
No worries
> 2. I noticed the welcome page changed. I think @kordless was pretty adamant on that being the page it is today, so I'll let you two hash that one out :)
The problem with the welcome page is the instructions are (a) for community, (b) far, far too long, (c) cover several distinct tasks.

> 3. I think you can get rid of the Enterprise path and just have those same folders for Community. Looking for feedback from @GRaska , @jrbrinlee1 , and @kcrodgers24 here, but it seems like we don't need a whole section for enterprise.
Ok, if that works for you all :)

> 4. I don't see anything under Concepts as of now, but we'll need that for pages today. Will that mirror cloud and community?
Concepts will happen once the content is restructured as this folder is for the high-level concepts not the product-specific info.
> 5. Cloud and Community seem to have different folder, but likely should be the same right? (i.e. Enterprise is missing data modeling, and cloud is missing API, etc.)
??  This was the structure outlined in the Confluence page which was designed to give users a single point of reference for help on the product they've got. Also, I've been told they're different products anyway, with different setup and config and similar but **not** identical functionality

> 6. SQL-Preview should come out of enterprise and go where SQL Guide is. We might have a Sql preview expand and a SQL expand. Most of the pages under SQL guide are actually a part of the preview (those have a note at the tope of the page)
Is it a generic feature for all the different products? That is, if I've paid cash for Cloud, or done a trial install with community, will this work?

> 7. There are a lot of pages missing right now, but every page should have a home before we can merge in. I know you just started this but just as a heads up. Let me know if you need help sorting
I haven't deleted any pages, and any stragglers are documented in DOCS-43 for reference :)

> 8. Once pages are there, links are.... DEAD. Drives the importance of let's get everything in the right place, then go in and address links. I'll help gather external links, but you can use the python script in the repo (dead link seeker) to find all the dead links. There are instructions in the README, but let me know if ya need some help on that.

I'll be looking at the hyperlinks today. A dead link seeker will potentially be of limited use, because **all** the links are dead right now :D
> 
> Sorry for the long comment, but hopefully this gets any other feedback out there! We'll keep iterating!
All good.

**Comment by kcrodgers24** [comment]
"I think you can get rid of the Enterprise path and just have those same folders for Community. Looking for feedback from @GRaska , @jrbrinlee1 , and @kcrodgers24 here, but it seems like we don't need a whole section for enterprise."
100% endorsed. Let's make it all Community.

**Comment by gthrone25** [comment]
> > @ljsinclair Putting notes below as they mostly relate to structure at this point:
> > 
> > 1. Great start! thanks for getting this out so quick
> >    No worries
> > 2. I noticed the welcome page changed. I think @kordless was pretty adamant on that being the page it is today, so I'll let you two hash that one out :)
> >    The problem with the welcome page is the instructions are (a) for community, (b) far, far too long, (c) cover several distinct tasks.
@kordless your time is running out
> > 3. I think you can get rid of the Enterprise path and just have those same folders for Community. Looking for feedback from @GRaska , @jrbrinlee1 , and @kcrodgers24 here, but it seems like we don't need a whole section for enterprise.
> >    Ok, if that works for you all :)
> 
> > 4. I don't see anything under Concepts as of now, but we'll need that for pages today. Will that mirror cloud and community?
> >    Concepts will happen once the content is restructured as this folder is for the high-level concepts not the product-specific info.
Sounds good
> > 5. Cloud and Community seem to have different folder, but likely should be the same right? (i.e. Enterprise is missing data modeling, and cloud is missing API, etc.)
> >    ??  This was the structure outlined in the Confluence page which was designed to give users a single point of reference for help on the product they've got. Also, I've been told they're different products anyway, with different setup and config and similar but **not** identical functionality
I can get on board with that. Would we not want that page to link to api docs?
> > 6. SQL-Preview should come out of enterprise and go where SQL Guide is. We might have a Sql preview expand and a SQL expand. Most of the pages under SQL guide are actually a part of the preview (those have a note at the tope of the page)
> >    Is it a generic feature for all the different products? That is, if I've paid cash for Cloud, or done a trial install with community, will this work?
Yes for the most part. It's the same as PQL and the other SQL.

> > 7. There are a lot of pages missing right now, but every page should have a home before we can merge in. I know you just started this but just as a heads up. Let me know if you need help sorting
> >    I haven't deleted any pages, and any stragglers are documented in DOCS-43 for reference :)
Gotcha. I more mean there are no pages under any cloud/community sub folders as of now. You maybe are rolling that into the linking work though :) 
> > 8. Once pages are there, links are.... DEAD. Drives the importance of let's get everything in the right place, then go in and address links. I'll help gather external links, but you can use the python script in the repo (dead link seeker) to find all the dead links. There are instructions in the README, but let me know if ya need some help on that.
> 
> I'll be looking at the hyperlinks today. A dead link seeker will potentially be of limited use, because **all** the links are dead right now :D
Sounds good!
> > Sorry for the long comment, but hopefully this gets any other feedback out there! We'll keep iterating!
> > All good.

**Comment by GRaska** [comment]
> "I think you can get rid of the Enterprise path and just have those same folders for Community. Looking for feedback from @GRaska , @jrbrinlee1 , and @kcrodgers24 here, but it seems like we don't need a whole section for enterprise." 100% endorsed. Let's make it all Community.

The only purpose for Enterprise path is our existing customers using the on-prem product. As long as @jrbrinlee1 feels they are covered/informed, then it makes sense to merge as much as possible.

**Comment by gthrone25** [comment]
@ljsinclair I just pushed a commit here that mirrors the changes I'm pushing into prod. Time to test if keeping this synced is gonna work or be a nightmare 🤞 . We can talk about it next week!

**Comment by ljsinclair** [comment]
I've now completed the work on DOCS-43, to:
* create new folders based on those in the restructure doc -- https://molecula.atlassian.net/wiki/spaces/DOCS/pages/1010499772/FeatureBase+documentation+issues+and+recommendations#Proposed-structure
* moved content from old structure to new
* changing URLs to point to new folders

FYI, the files in the conflict message have new URLs:
* `data-ingestion/cloud/streaming/` = `/cloud/cloud-data-ingestion/streaming`
* `data-modeling-guide/examples/` is now `/concepts/
* `data-querying/cloud/` is now `/cloud/query-cloud-data/`

File: _data/navigation.yml [hunk 1]
```diff
@@ -1,116 +1,43 @@
...
-  - title: Data Ingestion
-    category: data-ingestion
+      - title: Cloud setup
+        category: cloud-setup
+      - title: Cloud data ingestion
+        category: cloud-data-ingestion
+      - title: Cloud data modeling
+        category: cloud-data-modeling
+      - title: Query cloud data
+        category: query-cloud-data
```
Comment (gthrone25) [inline _data/navigation.yml:17]: @ljsinclair  To fix the cloud nav so that all the pages show when performing the build, change the category and add a subnav for streaming under ingest:
```
  - title: Cloud
    category: cloud
    subnav:
      - title: Cloud setup
        category: cloud-setup
      - title: Cloud data ingestion
        category: cloud-data-ingestion
        subnav:
          - title: Streaming
            category: streaming
      - title: Cloud data modeling
        category: cloud-data-modeling
      - title: Query cloud data
        category: query-cloud-data
```

Please note that the folder name "Cloud" must also change to "cloud" here. To be consistent, let's have all the folders stay lowercase, which propagates to navigation

**Comment by gthrone25** [comment]
@ljsinclair It will be hard to validate links until the nav file is finalized because I see a lot broken as of now. A couple other things I saw while I went through the nav file:
1. All of the PQL Guide pages are not under their proper folder (READ, WRITE, OTHER)
2. SQL Preview needs to move under SQL guide as it's own folder (or what you think is best outside of enterprise)
3. SQL Guide is still showing the SQL preview pages. They'll need to move under SQL preview once that's out of enterprise

**Comment by gthrone25** [comment]
@ljsinclair last thing! I have resolved conflicts and pushed them up :)

**Comment by ljsinclair** [comment]
Okay, so sorry about this extremely messy Pull request.
As of now, 2022-11-04, I've moved all the files to new folders, fixed all the URLs and links and done some *very* light work on fixing filenames and URLs.
You can see the help in staging by running thru this confluence page: 
https://molecula.atlassian.net/wiki/spaces/DOCS/pages/1010500163/FeatureBase+documentation+staging+test+environment

**Comment by gthrone25** [comment]
@ljsinclair I just tried to pull and build this branch and got an error that stems from this line in the index.html

`{% include /docs/get-support-source.md %}`

> jekyll 3.9.0 | Error:  Could not locate the included file '/docs/community/fbcom-install-links.md' in any of ["/Users/gregorythrone/Documents/Molecula/GitPull/documentation-43/_includes", "/Library/Ruby/Gems/2.6.0/gems/jekyll-theme-primer-0.6.0/_includes"]. Ensure it exists in one of those directories and is not a symlink as those are not allowed in safe mode

This is different than what's on the remote server though, so I'm not sure what is up to date. Please let me know. Things look very close on the remote server to me! Only a couple things I'm hoping you're ok with:
1. Thoughts on data modeling being a folder in concepts? Happy to wait on this but technically "data modeling, "time quantums and ttl" and "A true crime story" all are data modeling concepts
2.  Can we move the sql-preview page: http://3.133.140.233:4000/sql-preview/sql-preview to the top of the expand? (change the date)

**Comment by ljsinclair** [comment]
Fixed the broken include
Rearranged some content, particularly:
* Concepts
* Cloud ingestion, streaming
* Community setup
Mainly this involves:
* adding headings to make the content easier to find (before you begin prerequisites, next step to link content together logically)
* splitting out duplicate information (e.g., TTL, Time quantums)
* fixing datestamps so the content appears in the correct order
I also worked on the Crime story page, which will need a review to make sure it makes sense.
This one was converted to a Case study, and I rewrote long explanations because they were too hard to follow.

**Comment by gthrone25** [comment]
@ljsinclair I did notice a big issue with the home page (index.html) in that every single value for No/Yes is currently reversed. To fix this every No must become "Yes" and every "Yes" must become "No". For example, Cloud Click to Deploy is "Yes" and Community is "No". One other thought on this page is I feel we need at least a high level description of FeatureBase. Any reason not to have one? something similar to what most DBs have: https://clickhouse.com/docs/en/home/.

**Comment by gthrone25** [comment]
Seems Creating a database got moved to data ingestion and databases overview got moved to data modeling, and I'm wondering why? They need to be together, and I thought SETUP was the correct spot. That mirrors setup in community. You could make the argument to add a "Database" folder as well

**Comment by ljsinclair** [comment]
So, to summarise:
* moved the databases pages to /cloud-setup
* fixed links
* fixed homepage reversal of yes/no
* homepage summary > that's what the first link under Before you begin is for -- to direct the user to the biz page where they can get all the info (another of these "don't reinvent the wheel" things I keep going on about)
* restored original crime story
* changed datestamp on the things done today around community-data-ingestion
As discussed, I'm not touching this branch because it'll be merged soon.
And I'll be more circumspect about branching in future. Thanks for your help and patience

**Comment by gthrone25** [comment]
Closing due to merge conflicts. docs-43-new was used to merge all of these changes. Well done on all the work @ljsinclair !
